File: Ejercicio/Python/ejercicio.py
from random import randint
from math import sqrt, pow

class Ejercicio:
    def __init__(self):
        self.radio = 0
        self.height = 768
        self.width = 1024
        self.porcentaje = 0

    def setRadio(self):
        radioTemp = 0
        while (radioTemp < 50 or radioTemp > 600):
            radioTemp = int(input("Ingrese el radio (entre 50 y 600): "))
        self.radio = radioTemp

    def setPorcentaje(self):
        self.setRadio()
        suma = 0
        count = 0
        X = []
        Y = []
        while count < 1000:
            X.append(randint(0, self.width-1))
            Y.append(randint(0, self.height-1))
            circ = sqrt((pow(X[count]-(self.width/2),2))+(pow(Y[count]-(self.height/2),2)))
            if circ <= self.getRadio():
                suma += 1
            count += 1
        self.porcentaje = (suma / 1000) * 100

    def getRadio(self):
        return self.radio

    def getPorcentaje(self):
        return self.porcentaje

File: Ejercicio/C++/Ejercicio.cpp
#include <iostream>
#include <ctime>
#include <cmath>
#include "Ejercicio.h"

using namespace std;

void Ejercicio::setRadio() {
	int radioTemp;
	do {
		cout << "Ingrese el radio (entre 50 y 600): ";
		cin >> radioTemp;
	} while (radioTemp < 50 || radioTemp > 600);
	radio = radioTemp;
	cout << endl;
}

int Ejercicio::getRadio() {
	return radio;
}

void Ejercicio::setPorcentaje() {
	setRadio();
    srand(time(NULL));
	int suma = 0, circ, X[1000], Y[1000];
	for (int i = 0; i < 1000; i++) {
		X[i] = (rand() % width);
		Y[i] = (rand() % height);
		circ = sqrt((pow(X[i]-(width/2),2))+(pow(Y[i]-(height/2),2)));
		if (circ <= getRadio()) {
			suma++;
		}
	}
	porcentaje = (float(suma) / 1000.0) * 100.0;
}

float Ejercicio::getPorcentaje() {
    return porcentaje;
}